### PR TITLE
Features/retrieve usages and sources

### PR DESCRIPTION
**Pull request by 247benji**
1) Added view model for source documents. Added methods to retrieve usages and source documents.
2) Added unit tests for retrieving usages and sources.

### CODE REVIEW

**Comment by UKHOGreg** [comment]
> Does the order of the SourceDocuments matter? If so then there should be a test to reflect this.

I don't think so as we will likely be sorting them client-side.
If we end up sorting server side we shall make a test!

> I would like to see more/altered unit tests:
> * To verify that the `SourceViewModel` has been populated correctly

We can do that!

> * To ensure that only the correct ProcessId and status is picked up (just add some setup data with different processIds and statuses)

The status issue is one we were hoping to revist later in the development of the story, we absolutely intend to cover that with unit tests!

> * To ensure that if you add linked and database documents that they are both picked up

So a test that makes sure that both are added at the same time?
Sure, we can do that!

File: src/Portal/Portal.UnitTests/EditDatabaseTests.cs [hunk 1]
```diff
@@ -116,5 +117,180 @@ public void Test_CreateCarisProject_Throws_InvalidOperationException_When_No_Hpd
...
+
+            await _editDatabaseModel.OnGetAsync(ProcessId, "Assess");
+
+            //Expected value is 1 higher than added DatabaseDocumentStatus as an item
+            //is added to SourceDocuments in SetupForOnGetAsync()
+            Assert.AreEqual(3, _editDatabaseModel.SourceDocuments.Count);
+        }
+
+        [Test]
+        public async Task Test_Only_Documents_With_Matching_ProcessId_Are_Added()
```
Comment (dynamictulip) [inline src/Portal/Portal.UnitTests/EditDatabaseTests.cs:246]: Can we add a `LinkedDocument` and `DatabaseDocumentStatus` with a different `ProcessId` too please

File: src/Portal/Portal.UnitTests/EditDatabaseTests.cs [hunk 2]
```diff
@@ -116,5 +117,180 @@ public void Test_CreateCarisProject_Throws_InvalidOperationException_When_No_Hpd
...
+
+            await _editDatabaseModel.OnGetAsync(ProcessId, "Assess");
+
+            //Expected value is 1 higher than added DatabaseDocumentStatus as an item
+            //is added to SourceDocuments in SetupForOnGetAsync()
+            Assert.AreEqual(2, _editDatabaseModel.SourceDocuments.Count);
+        }
+
+        [Test]
+        public async Task Test_SourceViewModel_Has_Been_Populated()
```
Comment (dynamictulip) [inline src/Portal/Portal.UnitTests/EditDatabaseTests.cs:208]: The source view model is populated in 3 places so we need three tests

File: src/Portal/Portal.UnitTests/EditDatabaseTests.cs [hunk 3]
```diff
@@ -7,6 +7,7 @@
 using Microsoft.EntityFrameworkCore;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
+using Microsoft.Graph;
```
Comment (Rossall82) [inline src/Portal/Portal.UnitTests/EditDatabaseTests.cs:10]: Heh - don't think we need this using

Comment (UKHOGreg) [reply src/Portal/Portal.UnitTests/EditDatabaseTests.cs]: Thanks, removed!

**Pull request merged**